redash/tasks/queries.py

```diff
@@ -277,14 +280,46 @@ def get_table_sample_data(existing_columns, data_source_id, table_name, table_id
         column_examples
     )
     models.db.session.commit()
+    logger.info(u"task=get_table_sample_data state=finished table_name=%s runtime=%.2f",
+                table_name, time.time() - start_time)
+
 
+@celery.task(name="redash.tasks.get_all_table_samples")
+def get_all_table_samples(data_source_id):
+    TABLE_SAMPLE_LIMIT = 50
+    ds = models.DataSource.get_by_id(data_source_id)
+
+    if not ds.query_runner.configuration.get('samples', False):
+        return
+
+    DAYS_AGO = (
+        utils.utcnow() - datetime.timedelta(days=settings.SCHEMA_SAMPLE_REFRESH_FREQUENCY_DAYS))
+
+    # Find all existing tables that have an empty or old latest_sampling
+    tables_to_sample = TableMetadata.query.filter(
+        TableMetadata.exists.is_(True),
+        or_(
+            TableMetadata.latest_sampling.is_(None),
+            TableMetadata.latest_sampling < DAYS_AGO
+        )
+    ).limit(TABLE_SAMPLE_LIMIT).all()
+
+    for table in tables_to_sample:
+        get_table_sample_data.apply_async(
+            args=(ds.id, table.name, table.id),
+            queue=settings.SCHEMAS_REFRESH_QUEUE
+        )
+        table.latest_sampling = db.func.now()
+        models.db.session.add(table)
```

Comment:
It may make sense here to use a [Celery group](https://celery.readthedocs.io/en/latest/userguide/canvas.html#groups) to schedule the tasks in a group (which is executed in parallel) so the tasks are run in a timely fashion and scheduling happens in one bulk queuing:

```python
from celery import group
# ...
     tasks = []
     for table in tables_to_sample:
        tasks.append(
    		get_table_sample_data.signature(
                args=(ds.id, table.name, table.id),
                queue=settings.SCHEMAS_REFRESH_QUEUE
            )
        )
        table.latest_sampling = db.func.now()
        models.db.session.add(table)

    group(tasks).apply_async()
    models.db.session.commit()